diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -849,12 +862,16 @@ namespace diff_drive_controller
     limiter_lin_.limit(curr_cmd.lin, last0_cmd_.lin, last1_cmd_.lin, control_period);
     limiter_ang_.limit(curr_cmd.ang, last0_cmd_.ang, last1_cmd_.ang, control_period);
 
-    last1_cmd_ = last0_cmd_;
-    last0_cmd_ = curr_cmd;
-
     // Compute/Update wheels velocities after enforcing the acceleration limits:
-    left_velocity_limited  = (curr_cmd.lin - curr_cmd.ang * ws / 2.0)/wrl;
-    right_velocity_limited = (curr_cmd.lin + curr_cmd.ang * ws / 2.0)/wrr;
+    vl = curr_cmd.lin - curr_cmd.ang * ws / 2.0;
```

Comment:
We should definitely do a coupled (robot + joint space) velocity limits enforcement instead of this, but that would be a lot of changes. I'd start some discussion offline in that direction, but this is ok in the short-term. I can't guarantee though that enforcing wheel limits violates other limits. It'd likely violate the turning radius.

The fact that we're having issues with this also means the we're likely not setting velocity and acceleration limits properly in the URDF + param server.